### DISCUSSION TITLE
Re-introducing the GitHub Public Roadmap 🗺️

### DISCUSSION THREAD

**Discussion by github-staff** [community/community#145255] [Announcements]
## Re-introducing the GitHub Public Roadmap

We are excited to introduce the refreshed [GitHub public roadmap](https://github.com/orgs/github/projects/4247/views/1)! We have dedicated the last few months to an extensive holistic review of the GitHub product roadmap and are now excited to release this refreshed and improved version. This refresh will make it easier for you to follow our progress, double-click on upcoming features, and provide a more accurate reflection of GitHub’s development priorities over the next 12 months. 

## What’s new

We have added new releases to the project board and sorted the releases into `Product Focus Area` and by release phase. 

![roadmap_2](https://github.com/user-attachments/assets/12e56c5e-2a30-4987-9208-8f16c006ae32)

Each item on the board links to a detailed issue with more information about what we’re planning, why it’s valuable, and when we tentatively expect to deliver it. You can also watch the repository or subscribe to individual roadmap issues to be notified about changes directly within GitHub. 

![roadmap_3](https://github.com/user-attachments/assets/7ddc0082-7f8f-41cc-bf56-fe9e32520192)

As part of our efforts to align with our evolving product vision, we’ve refined the roadmap by updating issues and introducing new ones that better reflect our current direction. If an issue you were following was removed from the roadmap, it may resurface in a future update and we are adding details to the removed issues to help you understand what’s coming next.  

We remain committed to keeping the roadmap updated regularly and enhancing transparency through our [quarterly webinars](https://github.com/roadmap-webinar-series), which provide an in-depth, interactive way to stay connected to our product roadmap and latest releases.

## Share your feedback

If you have feedback about the roadmap refresh, feel free to engage with your GitHub representative or raise your thoughts in the comments. 👇  We plan on continuing to iterate and improve our roadmap as we hear your feedback! 📣

**Comment by hauntsaninja** [comment]
@ankneis I don't want to shoot the messenger, but since you explicitly recommended continuing the conversation here:

- Is there a place I can subscribe to for this specific feature like https://github.com/github/roadmap/issues/347 or https://github.com/github/roadmap/issues/552 where I get notified when it is implemented?
- You mentioned posting a sentence or two of extra detail here: https://github.com/github/roadmap/issues/347#issuecomment-2501845491 . I was a little confused by the claim "Code review improvements remain a key priority for us" given that as far as I can tell, there is only one issue in the above roadmap related to code review, which is an AI reviewing thing (that is maybe enterprise only?)

This situation was a nice opportunity to go over the roadmap though, e.g. I'm excited to see what https://github.com/github/roadmap/issues/1036 will look like, especially the more advanced search.

Finally, some unsolicited advice... The language around this rollout has been pretty corporate and vague — while people will always be unhappy to see their feature requests officially deprioritised, I'd expect you'd have gotten a better response either a) using more direct language, b) not explicitly soliciting feedback you didn't actually want to hear (i.e. just keep the announcement post locked)

**Reply by AA-Turner** [reply]
Hi @ankneis,

To echo @hauntsaninja, both of the mentioned (missing) features are frequent annoyances when using GitHub, and it seems a loss that there aren't tracking items for them any more.

Code review is one of the most frequent interactions I (and I would posit many other maintainers of large projects) have with GitHub, and it feels like a real opportunity loss. The AI-type features mentioned can be emulated using external tools, but these are limitations of the platform itself which we can't, as end-users, work-around.

I do hope to hear good news in the future, and thank you for the invitation to write on this forum.

A

**Reply by samus-aran** [reply]
Hi @hauntsaninja (& @AA-Turner too)

A lot of great points! I can only speak to the community communication side of things, but we hear you loud and clear—there’s definitely room for us to improve how we communicate.

Your feedback is incredibly valuable to us, and we absolutely want to hear it. We locked the roadmap discussion to guide users here, where we have a dedicated team of community managers and PMs actively monitoring the forums. This setup allows us to engage better and respond more quickly than we could in the roadmap discussions.

As for tracking closed roadmap items, you can still stay updated by subscribing to them. 

<img src="https://github.com/user-attachments/assets/ef0e1f87-2ce3-4d4f-90ec-bc6bc78ee929" width="50%" />

This way, you’ll be notified about any status updates or new comments. On top of that, we’re putting together a list of focused feedback discussions tied to specific roadmap items. This will give you a chance to join those conversations and share your thoughts directly with our team.

I’ve pulled the two items you mentioned, and we’ll review our previous communication to ensure we’re guiding users more directly. 👍
- https://github.com/github/roadmap/issues/347
    - [Allow PR reviewers to comment on *unchanged* files](https://github.com/orgs/community/discussions/9099)
- https://github.com/github/roadmap/issues/552
    - [Top level PR comment thread](https://github.com/orgs/community/discussions/8469)


Really appreciate your straightforward feedback—thank you! ❤️

**Reply by hauntsaninja** [reply]
Thank you so much! Great to hear the closed roadmap issues will still be updated if reprioritised. And thank you for linking to the equivalent discussions on community, I've subscribed to those as well :-)

**Comment by richard-sim** [comment]
Adding yet-another vote how massively disappointing the removal of https://github.com/github/roadmap/issues/347 and https://github.com/github/roadmap/issues/552 from the roadmap is.

347 is especially disappointing, as it's incredibly common to have to point out omissions in PR's that need to be addressed, and since they're omissions, by definition they aren't edited lines, so can't be commented on!

552 is a little less disappointing to me, as I never, ever use PR-level comments because they're so useless. That might be a sign that the issue should be addressed... just say'n.

We're (almost) all dev's here - we understand product prioritization, and how hard it can be. But please listen to your users here, as you're putting features that should be considered core product functionality in the ice box in order to prioritize glossy marketing features. Yes, there needs to be a balance. This is not it.

**Reply by pboling** [reply]
I totally understand the need to pad the stock price with AI buzz, but important issues like the following need to be addressed:

- addressing the _systemic racism_ of the [attempt to add admonitions to Markdown](https://github.com/orgs/community/discussions/16925#discussioncomment-12241696)
- fixing GitHub actions with a proper "allow-failure" feature that many other CI tools have. The lack of this feature often results in [broken software being accidentally released](https://github.com/actions/runner/issues/2347), with various examples in the [related threads](https://github.com/orgs/community/discussions/15452#discussioncomment-12537084)
- [Literally lying to end users](https://github.com/orgs/community/discussions/37117) who auth via GitHub about the permissions they are granting

I don't expect new responses from GitHub on any of these issues. My opinion of GitHub could not drop lower, and I am putting in many hours of effort to ensure I use as few GitHub services as possible.

**Comment by fajpunk** [comment]
I too am very dissapointed that  github/roadmap#347 and github/roadmap#552 are not going to be implemented.

**Comment by FrancescoBonizzi** [comment]
Also https://github.com/github/roadmap/issues/824 (More control over required status checks for pull requests using merge queue) has been abandoned. Sad.

This is a common use case where certain PR checks should be skipped based on specific filters.

In our monorepo, we have three main projects:

- Mobile App
- Web
- Backend

We use three PR action checks, each filtered by folder. As a safety measure, it’s crucial to block merging into the main branch if the relevant check hasn’t passed. However, without this feature, we’re forced to leave all checks optional, which compromises safety.

The ideal solution would be to **just send a skip signal to the required check when a pr action is skipped due to path filters**!

**Comment by mgu-voyado** [comment]
I know echoing the same thing over and over doesn't add much meaningful conversation, but since feedback is better than silence I also want to highlight [github/roadmap#347](https://github.com/github/roadmap/issues/347) as something very dear to me and I hope it can return to the roadmap sooner rather than later :)

**Comment by showhand168** [comment]
it is good.

**Reply by showhand168** [reply]
i am so execited here.i get here as the first time.

**Comment by Danielku15** [comment]
I'd like to revive the discussion around https://github.com/github/roadmap/issues/930. It was quite a clear backlog item and considering the deprecation of artifacts v3, and the performance impact of using it, I am voting to bring this back to the new roadmap. 

I cannot judge the remaining efforts to actually bring the new artifact backend to GHES but Github.com has it today, so the software should be capable of it. I also assume it is cumbersome for you to maintain the old artifact backend as part of GHES for all the releases. 

I hope to see some feedback and insights on the plans for artifact v4 support. TBH, we currently do not care much about any of the planned items for GHES for Q1 and Q2. But the artifact v3 costs us time&money every day.

**Reply by apmorton** [reply]
Strongly agree - this is such a huge pain point and we're actively having to work around terrible performance in artifact v3 at our company.

**Reply by AnthonyPaulO** [reply]
I don't understand why artifacts v4 is not even on the roadmap; been waiting years for this, what are GHES users supposed to do?

**Reply by pr11me** [reply]
Besides performance, other actions are relying on artifacts v4+ so that one has to use old versions of these on GHES. Example: https://github.com/gradle/actions/blob/8379f6a1328ee0e06e2bb424dadb7b159856a326/docs/dependency-submission.md, forced to use v3 instead of v5!

**Reply by Shanayara** [reply]
What is even so difficult about bringing the performance improvements to v3, or making v4+ available to GHES? The main problem with the performance literally is just about some zipping parameters, from what I've read in the issues. Fixable within a short time, it seems to me. The only explanation for me is that GitHub actually hates us using GHES for artifacts somehow, or this would long have been implemented.

**Reply by KJMascarenhas** [reply]
In addition to the performance improvements, we would also like to use the features for downloading artifacts from a different workflow run or even from a different repo.
Please try to prioritize support for GHES in the latest version of the download-artifact Action

**Reply by pdewilde** [reply]
Without v4 it means I cannot use the same workflow on github.com and GHES. The small differences add up. Every time I use `gh` in a command I need to remember to use GH_ENTERPRISE_TOKEN and GH_HOST, rather than GH_TOKEN. I need to remember to switch my runner type. I can't even use v3 everywhere because it's no longer supported in `.com`.

**Comment by dloetzke** [comment]
Can we bring github/roadmap#636 back onto the roadmap?
If you build reusable workflows as described in that issue, it is very frustrating to find out that you can't use secrets from the called repository although the workflow lives there. Additionally, having to define the exact same secrets in all caller repositories, only to pass them to the very same reusable workflow, is quite a lot of extra work at no real benefit (instead, it seems more like a security concern to me).

**Reply by audunsolemdal** [reply]
I found it a bit strange that this issue was not included in the new roadmap. This is a good security improvement over existing solution.

**Comment by zhuoyang** [comment]
please bring https://github.com/github/roadmap/issues/578 back onto the roadmap, we are a Java house and using Github Packages Maven has been a PITA.

**Reply by SilverPreeceSOF** [reply]
I agree, I personally think it's bonkers to have removed this from the roadmap. GitHub has _serious_ problems supporting an enterprise CI/CD setup and leaving critical issues on the roadmap for several years, kicking them down the road before removing them entirely is incredibly disheartening.

We have Maven applications in GitHub, so it only makes sense to use GitHub Packages and its handy Maven front-end to hold our built artifacts. It's right there, it (supposedly) integrates with everything else, so why not? The "why not" becomes extremely obvious when you realise that authentication with it requires a personal access token, and no other kind of token is valid. 

Do you have multiple Maven projects that you want to share that configuration with? Of course you do! Why else would you be using GitHub Packages? Well, best get ready to manually copy that same token into _**every single repository**_ because secret management in GitHub remains absolutely atrocious. You can either declare every secret and variable for every application at the organisation level, which sounds nightmarish, or you can laboriously update the same configuration in a hundred different locations, which is also nightmarish.

A Microsoft rep straight-up told us that "most customers roll their own UI". Are you kidding? We are simultaneously told that Azure DevOps, a solution that seems infinitely more mature to my layman eyes, is EOL and should not be used. Why would Microsoft push its users away from a feature complete solution to one that forces them to build custom tooling on top of it to perform basic configuration tasks?

**Reply by pdefreitas** [reply]
GitHub packages Maven users are second class citizens for a couple of years since this feature is available for other package managers. There's no way to do proper RBAC and currently this behavior just breaks Dependabot integration since even if you use a PAT as a secret, it will not be injected into GitHub actions workflows triggered by Dependabot. This means that your CI/CD flows won't be able to download your organization wide packages via Dependabot.

**Comment by Brakistad** [comment]
Hi GitHub and @github-staff,

I appreciate your efforts to refine the roadmap for better clarity. In the past, I've noticed some issues being postponed, so this cleanup is a positive step.

However, I have a question: In [Epic issue #1116](https://github.com/github/roadmap/issues/1116), several features are listed as in progress. One feature I've been following closely is [Issue #558](https://github.com/github/roadmap/issues/558), which seems to have been removed from the public roadmap. Given its inclusion in the epic, I'm curious about its current status. Was it deprioritized, or was its removal an oversight during the cleanup?

Thanks for any insights you can provide!

**Comment by aarongable** [comment]
One of the issues removed from the roadmap was https://github.com/github/roadmap/issues/281, tracking creating custom workflows within GitHub Projects. If this feature has been deprioritized or shelved, that's okay -- I understand how engineering prioritization works, and the necessary resources are needed elsewhere right now.

However, that particular feature is not just still [highly](https://github.com/orgs/community/discussions/53705) [requested](https://github.com/orgs/community/discussions/45653) on the Community board, it is _explicitly advertised_ on the [GitHub Issues and Projects homepage](https://github.com/features/issues):

<img width="1285" height="773" alt="image" src="https://github.com/user-attachments/assets/70c39cbe-2d36-49c5-89b9-a2cde4366863" />

Can we either get confirmation that this feature is still planned, or get the advertisement of the feature removed from the Projects page so we know to stop waiting for it?

**Comment by hfhbd** [comment]
The roadmap is outdated, many features that are listed for Juli to September are not shipped, mostly [Fine-grained PATs](https://github.com/github/roadmap/issues/790) or [Immutable Actions](https://github.com/github/roadmap/issues/1103).

And [Packages support for fine-grained PATs](https://github.com/github/roadmap/issues/558) is still open but removed from the project.

When do you plan to update the roadmap?
